internal/data/generated_slices.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Code generated by slice_generator_main.go. DO NOT EDIT.
+// To regenerate this file run `go run internal/data/internal/slice_generator_main.go`.
+
+package data
+
+import (
+	"sort"
+
+	otlpmetrics "github.com/open-telemetry/opentelemetry-proto/gen/go/metrics/v1"
+)
+
+// MetricsSlice logically represents a slice of Metric.
+//
+// Must use NewMetricsSlice function to create new instances.
+// Important: zero-initialized instance is not valid for use.
+type MetricsSlice struct {
+	orig *[]*otlpmetrics.Metric
+}
+
+// NewMetricsSlice creates a MetricsSlice with `len` empty elements.
+//
+// Caller must initialize all the allocated elements.
```

Comment:
Comment is unclear. What does "initialize" mean?

Reply:
They are empty messages.

Reply:
I meant that the caller has to set the fields, otherwise they are just empty messages.

Reply:
Isn't that what one would normally expect from a New* method? What else can they be? I was confused by the comment. I thought I will need to do some additional "initialization" before I can use the messages. But empty messages are completely valid messages. I think the comment is unnecessary.

Reply:
ok, will remove it.